Managers/ConsoleMan.h

```diff
@@ -1,314 +1,198 @@
+		/// <returns>The ratio of the screen that is covered.</returns>
+		float GetConsoleScreenSize() const { return m_ConsoleScreenRatio; }
+
+		/// <summary>
+		/// Sets how much of the screen that the console should cover when opened.
+		/// </summary>
+		/// <param name="screenRatio">The ratio of the screen to cover. 0 - 1.0.</param>
+		void SetConsoleScreenSize(float screenRatio = 0.3F);
+#pragma endregion
+
+#pragma region Logging
+		/// <summary>
+		/// Writes all the input strings to a log in the order they were entered.
+		/// </summary>
+		/// <param name="filePath">The filename of the file to write to.</param>
+		void SaveInputLog(std::string filePath);
+
+		/// <summary>
+		/// Writes the entire console buffer to a file.
+		/// </summary>
+		/// <param name="filePath">The filename of the file to write to.</param>
+		void SaveAllText(std::string filePath);
+
+		/// <summary>
+		/// Clears all previous input.
+		/// </summary>
+		void ClearLog();
+#pragma endregion
+
+#pragma region Concrete Methods
+		/// <summary>
+		/// Prints a string into the console.
+		/// </summary>
+		/// <param name="stringToPrint">The string to print.</param>
+		void PrintString(std::string stringToPrint);
+
+		/// <summary>
+		/// Opens the console and prints the shortcut help text. This is called from UInputMan when F1 is pressed.
+		/// </summary>
+		void ShowShortcuts();
+
+		/// <summary>
+		/// Updates the state of this ConsoleMan. Supposed to be done every frame before drawing.
+		/// </summary>
+		void Update();
+
+		/// <summary>
+		/// Draws this ConsoleMan's current graphical representation to a BITMAP of choice.
+		/// </summary>
+		/// <param name="targetBitmap">A pointer to a BITMAP to draw on.</param>
+		void Draw(BITMAP *targetBitmap);
+#pragma endregion
+
+#pragma region Class Info
+		/// <summary>
+		/// Gets the class name of this object.
+		/// </summary>
+		/// <returns>A string with the friendly-formatted type name of this object.</returns>
+		virtual const std::string & GetClassName() const { return c_ClassName; }
+#pragma endregion
+
+	protected:
+
+		static const std::string c_ClassName; //!< A string with the friendly-formatted type name of this object.
+
+		/// <summary>
+		/// //!< Enumeration for console states when enabling/disabling the console. NOTE: This can't be lower down because m_ConsoleState relies on this definition.
+		/// </summary>
+		enum ConsoleState { ENABLING = 0, ENABLED, DISABLING, DISABLED };
+
+		ConsoleState m_ConsoleState; //!< Current state of the console.
+		bool m_ReadOnly; //!< Read-only mode where console text input is disabled and controller input should be preserved.
+		float m_ConsoleScreenRatio; //!< The ratio of the screen that the console should take up, from 0.1 to 1.0 (whole screen).
+
+		GUIScreen *m_GUIScreen; //!< GUI Screen for use by the in-game GUI.
+		GUIInput *m_GUIInput; //!< GUI Input controller.
+		GUIControlManager *m_GUIController; //!< Manager of the console GUI elements.
```

Comment:
Can you rename this member name to match the object name (i.e. m_GUIControllManager)?